File: airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/JdbcSource.java
/*
 * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
 */

package io.airbyte.integrations.source.jdbc;

import com.fasterxml.jackson.databind.JsonNode;
import io.airbyte.db.factory.DatabaseDriver;
import io.airbyte.db.jdbc.JdbcUtils;
import io.airbyte.db.jdbc.streaming.AdaptiveStreamingQueryConfig;
import io.airbyte.integrations.base.IntegrationRunner;
import io.airbyte.integrations.base.Source;
import java.sql.JDBCType;
import java.util.Set;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class JdbcSource extends AbstractJdbcSource<JDBCType> implements Source {

  private static final Logger LOGGER = LoggerFactory.getLogger(JdbcSource.class);

  public JdbcSource() {
    super(DatabaseDriver.POSTGRESQL.getDriverClassName(), AdaptiveStreamingQueryConfig::new, JdbcUtils.getDefaultSourceOperations());
  }

  // no-op for JdbcSource since the config it receives is designed to be use for JDBC.
  @Override
  public JsonNode toDatabaseConfig(final JsonNode config) {
    return config;
  }

  @Override
  public Set<String> getExcludedInternalNameSpaces() {
    return Set.of("information_schema", "pg_catalog", "pg_internal", "catalog_history");
  }

  public static void main(final String[] args) throws Exception {
    final Source source = new JdbcSource();
    LOGGER.info("starting source: {}", JdbcSource.class);
    new IntegrationRunner(source).run(args);
    LOGGER.info("completed source: {}", JdbcSource.class);
  }

}

File: airbyte-cdk/java/airbyte-cdk/db-sources/src/main/kotlin/io/airbyte/cdk/integrations/source/jdbc/JdbcSource.kt
/*
 * Copyright (c) 2023 Airbyte, Inc., all rights reserved.
 */
package io.airbyte.cdk.integrations.source.jdbc

import com.fasterxml.jackson.databind.JsonNode
import io.airbyte.cdk.db.factory.DatabaseDriver
import io.airbyte.cdk.db.jdbc.JdbcUtils
import io.airbyte.cdk.db.jdbc.streaming.AdaptiveStreamingQueryConfig
import io.airbyte.cdk.integrations.base.IntegrationRunner
import io.airbyte.cdk.integrations.base.Source
import io.github.oshai.kotlinlogging.KotlinLogging
import java.sql.JDBCType
import java.util.function.Supplier

private val LOGGER = KotlinLogging.logger {}

class JdbcSource :
    AbstractJdbcSource<JDBCType>(
        DatabaseDriver.POSTGRESQL.driverClassName,
        Supplier { AdaptiveStreamingQueryConfig() },
        JdbcUtils.defaultSourceOperations
    ),
    Source {
    // no-op for JdbcSource since the config it receives is designed to be use for JDBC.
    override fun toDatabaseConfig(config: JsonNode): JsonNode {
        return config
    }

    override val excludedInternalNameSpaces: Set<String>
        get() = setOf("information_schema", "pg_catalog", "pg_internal", "catalog_history")

    companion object {

        @Throws(Exception::class)
        @JvmStatic
        fun main(args: Array<String>) {
            val source: Source = JdbcSource()
            LOGGER.info { "starting source: ${JdbcSource::class.java}" }
            IntegrationRunner(source).run(args)
            LOGGER.info { "completed source: ${JdbcSource::class.java}" }
        }
    }
}
